File: src/addon365.Web.API/ClientApp/src/app/models/lead.ts
import { BusinessContact } from "./business-contact";
import { LeadSource } from "./lead-source";
import { LeadStatusHistory } from "./lead-status-history";

export class Lead {
  id: string;
  sourceId: string;
  contact: BusinessContact;
  
  source: LeadSource;
  history: Array<LeadStatusHistory>;
}

File: src/DotNet/addon365.Erp.DataEntity/Crm/Lead.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace addon365.Database.Entity.Crm
{
    public class Lead : BaseEntityWithLogFields
    {
        public string Identifier { get; set; }
        public BusinessContact Contact { get; set; }
        public Guid SourceId { get; set; }

        [ForeignKey("SourceId")]
        public LeadSource Source { get; set; }

        public Guid CurrentLeadStatusId { get; set; }

        public IList<LeadStatusHistory> History { get; set; }
    }
}
